File: backend/src/main/java/com/supermarket/backend/model/Shift.java
package com.supermarket.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "shifts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Shift {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private LocalTime startTime;

    @Column(nullable = false)
    private LocalTime endTime;

    private String shiftType; // MORNING, AFTERNOON, EVENING

    private String register; // e.g., Register 04 (for cashiers)

    private boolean completed;
}

File: frontend/lib/models/shift.dart
class Shift {
  final int id;
  final DateTime date;
  final String startTime;
  final String endTime;
  final String shiftType;
  final String? register;
  final bool completed;

  Shift({
    required this.id,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.shiftType,
    this.register,
    required this.completed,
  });

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      id: json['id'] as int,
      date: DateTime.parse(json['date'] as String),
      startTime: (json['startTime'] as String).substring(0, 5),
      endTime: (json['endTime'] as String).substring(0, 5),
      shiftType: json['shiftType'] as String,
      register: json['register'] as String?,
      completed: json['completed'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'date': date.toIso8601String().substring(0, 10),
      'startTime': startTime,
      'endTime': endTime,
      'shiftType': shiftType,
      'register': register,
      'completed': completed,
    };
  }
}
